types/pg_hba_rule.pp

```diff
@@ -0,0 +1,14 @@
+# @summary type for all parameters in the postgresql::server::hba_rule defined resource
+# @see https://github.com/puppetlabs/puppetlabs-postgresql/blob/main/manifests/server/pg_hba_rule.pp
+type Postgresql::Pg_hba_rule = Struct[{
+    Optional[description]        => String,
+    type                         => Postgresql::Pg_hba_rule_type,
```

Comment:
Also optional - to be more precise: If an empty hash is OK, all keys must be Optional.